cpp/include/raft/stats/mean_center.cuh

```diff
@@ -78,6 +115,42 @@ void meanAdd(Type* out,
   detail::meanAdd<Type, IdxType, TPB>(out, data, mu, D, N, rowMajor, bcastAlongRows, stream);
 }
 
+/**
+ * @brief Add the input matrix wrt its mean
+ * @tparam Type the data type
+ * @tparam idx_t index type
+ * @tparam layout_t Layout type of the input matrix.
+ * @tparam TPB threads per block of the cuda kernel launched
+ * @param[in]  handle the raft handle
+ * @param[in]  data input matrix
+ * @param[in]  mu the mean vector
+ * @param[out] out the output mean-centered matrix
+ * @param[in]  bcast_along_rows whether to broadcast vector along rows or columns
+ */
+template <typename value_t, typename idx_t, typename layout_t>
+void mean_add(const raft::handle_t& handle,
+              raft::device_matrix_view<const value_t, idx_t, layout_t> data,
+              raft::device_vector_view<const value_t, idx_t> mu,
+              raft::device_matrix_view<value_t, idx_t, layout_t> out,
+              bool bcast_along_rows)
+{
+  static_assert(
+    std::is_same_v<layout_t, raft::row_major> || std::is_same_v<layout_t, raft::col_major>,
+    "Data layout not supported");
+  auto mean_vec_size = bcast_along_rows ? data.extent(1) : data.extent(0);
+  RAFT_EXPECTS(out.size() == data.size(), "Size mismatch");
+  RAFT_EXPECTS(mean_vec_size == mu.extent(0), "Size mismatch betwen data and mu");
```

Comment:
I've been taking `bcast_along_rows` to mean that the vector must be the number of columns if `bcast_along_rows==false` (e.g bcast_across_rows) and the vector must be the number of rows if `bcast_along_rows==true`.